askama_derive/src/lib.rs

```diff
@@ -21,26 +21,29 @@ use input::{Print, TemplateArgs, TemplateInput};
 #[proc_macro_derive(Template, attributes(template))]
 pub fn derive_template(input: TokenStream) -> TokenStream {
     let ast = syn::parse::<syn::DeriveInput>(input).unwrap();
-    match build_template(&ast) {
-        Ok(source) => source.parse().unwrap(),
-        Err(e) => {
-            let mut e = e.into_compile_error();
-            if let Ok(source) = build_skeleton(&ast) {
-                let source: TokenStream = source.parse().unwrap();
-                e.extend(source);
-            }
-            e
-        }
-    }
+    let e = match build_template(&ast) {
+        Ok(source) => return source.parse().unwrap(),
+        Err(e) => e,
```

Comment:
Let's do `e.into_compile_error()` here (no type annotation needed), and bind it to `tokens`.